components/style/custom_properties.rs

```diff
@@ -495,11 +495,45 @@ impl<'a> CustomPropertiesBuilder<'a> {
             return;
         }
 
+        // Check if we can avoid cloning the inherited CustomPropertiesMap.
+        match specified_value {
```

Comment:
Maybe this would be a bit nicer with a:

```rust
fn value_may_affect_style(&self, name: &Name, value: &DeclaredValue<Arc<SpecifiedValue>>) -> bool;
```

function?

That way `cascade` would read nicer, I think, and you don't need the extra scope.